### ISSUE TITLE
Error on saving changes in api

### ISSUE THREAD

**Issue by devsecur** [strapi/strapi#4875]
**Describe the bug**
After making some changing on any api in the application and on saving, I get the following error:

```
strapi_1  | [2020-01-02T07:11:32.419Z] error TypeError: tmpComponents.values is not a function
strapi_1  |     at Object.writeFiles (/srv/app/node_modules/strapi-plugin-content-type-builder/services/schema-builder/index.js:165:39)
strapi_1  |     at Object.editContentType (/srv/app/node_modules/strapi-plugin-content-type-builder/services/ContentTypes.js:127:17)
strapi_1  |     at Object.updateContentType (/srv/app/node_modules/strapi-plugin-content-type-builder/controllers/ContentTypes.js:92:50)
strapi_1  |     at async /srv/app/node_modules/strapi/lib/middlewares/router/utils/routerChecker.js:83:22
strapi_1  |     at async module.exports (/srv/app/node_modules/strapi-plugin-users-permissions/config/policies/permissions.js:40:14)
strapi_1  |     at async /srv/app/node_modules/strapi-utils/lib/policy.js:118:7
strapi_1  |     at async evalmachine.<anonymous>:70:11
```

**Steps to reproduce the behavior**
I'm not totally sure, the last thing I did was a hasmany connection between two api "map" and "pass":

```
strapi_1  | [2020-01-02T07:03:34.011Z] debug GET /content-manager/content-types (56 ms) 200
strapi_1  | [2020-01-02T07:03:34.104Z] debug GET /content-type-builder/components (58 ms) 200
strapi_1  | [2020-01-02T07:03:34.110Z] debug GET /content-type-builder/content-types (38 ms) 200
strapi_1  | [2020-01-02T07:06:08.874Z] info File changed: /srv/app/api/scene/models/scene.settings.json
strapi_1  | [2020-01-02T07:06:08.875Z] info File changed: /srv/app/api/pass/models/pass.settings.json
strapi_1  | [2020-01-02T07:06:08.879Z] debug POST /content-type-builder/content-types (252 ms) 201
strapi_1  | [2020-01-02T07:06:08.882Z] info The server is restarting
```

After saving, restart and changing any other api, I run into the error shown above

**Expected behavior**
Changes should be saved

**Screenshots**
<img width="1552" alt="Bildschirmfoto 2020-01-02 um 08 16 09" src="https://user-images.githubusercontent.com/32730503/71655671-29260f00-2d38-11ea-99b8-b68e76f308c6.png">

**System**
- Node.js version: From official docker image
- NPM version: From official docker image
- Strapi version: v3.0.0-beta.18.3
- Database: pg
- Operating system: From official docker image

**Comment by derrickmehaffy** [comment]
Can you please open a bug report on the docker repo: https://github.com/strapi/strapi-docker

Marking this one as closed for now.

**Comment by devsecur** [comment]
Found the error - I created an api called map - this leads to a new class "Map". In strapi-plugin-content-type-builder/services/schema-builder/index.js tmpComponents is defined as "Map" which is normally a list type, but in my case, it is the name of an api. But this class don't have the method "values" as Map does, so it throws an error. I see multiple solutions:

* Verify not to use reserved names for apis
* Make sure the api class name gets some kind of prefix
* Make sure the type of tmpComponets is selected right

This has nothing to do with the docker image but with strapi or the content builder plugin
